pinot-common/src/main/java/org/apache/pinot/common/minion/MinionTaskMetadataUtils.java

```diff
@@ -51,27 +64,46 @@ public static ZNRecord fetchTaskMetadata(HelixPropertyStore<ZNRecord> propertySt
   }
 
   /**
-   * Deletes the ZNRecord for the given minion task and tableName, from MINION_TASK_METADATA/taskName/tableNameWthType
+   * Deletes the ZNRecord for the given minion task and tableName, from both the new path
+   * MINION_TASK_METADATA/${tableNameWthType}/${taskType} and the old path
+   * MINION_TASK_METADATA/${taskType}/${tableNameWthType}.
    */
   public static void deleteTaskMetadata(HelixPropertyStore<ZNRecord> propertyStore, String taskType,
       String tableNameWithType) {
-    String path = ZKMetadataProvider.constructPropertyStorePathForMinionTaskMetadata(taskType, tableNameWithType);
-    if (!propertyStore.remove(path, AccessOption.PERSISTENT)) {
+    String newPath = ZKMetadataProvider.constructPropertyStorePathForMinionTaskMetadataNewFormat(
+        taskType, tableNameWithType);
+    String oldPath = ZKMetadataProvider.constructPropertyStorePathForMinionTaskMetadata(
+        taskType, tableNameWithType);
+    if (!propertyStore.remove(newPath, AccessOption.PERSISTENT)
```

Comment:
We want to remove from both if both of them exist. Current implementation will short-circuit the second part

Reply:
good catch!